source/adapters/cuda/memory.cpp

```diff
@@ -398,11 +398,109 @@ UR_APIEXPORT ur_result_t UR_APICALL urMemImageCreate(
-                                                      void *, size_t *) {
-  return UR_RESULT_ERROR_UNSUPPORTED_FEATURE;
+UR_APIEXPORT ur_result_t UR_APICALL urMemImageGetInfo(ur_mem_handle_t hMemory,
+                                                      ur_image_info_t propName,
+                                                      size_t propSize,
+                                                      void *pPropValue,
+                                                      size_t *pPropSizeRet) {
+  UR_ASSERT(hMemory->isImage(), UR_RESULT_ERROR_INVALID_MEM_OBJECT);
+
+  auto Context = hMemory->getContext();
+
+  ScopedContext Active(Context);
+  UrReturnHelper ReturnValue(propSize, pPropValue, pPropSizeRet);
+
+  try {
+    CUDA_ARRAY3D_DESCRIPTOR ArrayInfo;
+
+    UR_CHECK_ERROR(cuArray3DGetDescriptor(
+        &ArrayInfo, std::get<SurfaceMem>(hMemory->Mem).getArray()));
+
+    const auto cuda2urFormat = [](CUarray_format CUFormat,
+                                  ur_image_channel_type_t *ChannelType) {
+      switch (CUFormat) {
+      case CU_AD_FORMAT_UNSIGNED_INT8:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT8;
+        break;
+      case CU_AD_FORMAT_UNSIGNED_INT16:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT16;
+        break;
+      case CU_AD_FORMAT_UNSIGNED_INT32:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_UNSIGNED_INT32;
+        break;
+      case CU_AD_FORMAT_SIGNED_INT8:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_SIGNED_INT8;
+        break;
+      case CU_AD_FORMAT_SIGNED_INT16:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_SIGNED_INT16;
+        break;
+      case CU_AD_FORMAT_SIGNED_INT32:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_SIGNED_INT32;
+        break;
+      case CU_AD_FORMAT_HALF:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_HALF_FLOAT;
+        break;
+      case CU_AD_FORMAT_FLOAT:
+        *ChannelType = UR_IMAGE_CHANNEL_TYPE_FLOAT;
+        break;
+      default:
+        return UR_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT;
+      }
+      return UR_RESULT_SUCCESS;
+    };
+
+    const auto cudaFormatToElementSize = [](CUarray_format CUFormat,
+                                            size_t *Size) {
+      switch (CUFormat) {
+      case CU_AD_FORMAT_UNSIGNED_INT8:
+      case CU_AD_FORMAT_SIGNED_INT8:
+        *Size = 1;
+        break;
+      case CU_AD_FORMAT_UNSIGNED_INT16:
+      case CU_AD_FORMAT_SIGNED_INT16:
+      case CU_AD_FORMAT_HALF:
+        *Size = 2;
+        break;
+      case CU_AD_FORMAT_UNSIGNED_INT32:
+      case CU_AD_FORMAT_SIGNED_INT32:
+      case CU_AD_FORMAT_FLOAT:
+        *Size = 4;
+        break;
+      default:
+        return UR_RESULT_ERROR_UNSUPPORTED_IMAGE_FORMAT;
+      }
+      return UR_RESULT_SUCCESS;
+    };
+
+    switch (propName) {
+    case UR_IMAGE_INFO_FORMAT:
+      ur_image_channel_type_t ChannelType;
+      UR_RETURN_ON_FAILURE(cuda2urFormat(ArrayInfo.Format, &ChannelType));
```

Comment:
is there a benefit to switching to this instead of continuing to use UR_CHECK_ERROR?

Reply:
UR_CHECK_ERROR uses aborts which we are trying to eliminate. The addition of this macro is a cherry-pick from https://github.com/oneapi-src/unified-runtime/pull/1127. Essentially I'm just trying to make @martygrant life a bit easier so that he doesn't have to redesign this entrypoint on his PR.

Reply:
sorry I mean like within this entry point, we're already using UR_CHECK_ERROR above and since we're set up to catch the exception I don't think it can abort unless you set an env var specifically telling it to

Reply:
You are correct that we are not going to abort() in here. I forgot that an env var needs to be setup for that.

I'm honestly not sure of what is the correct style for errors. Where is it acceptable to throw (UR_CHECK_ERROR) and when should we return an error code (UR_RETURN_ON_FAILURE). 

The style I'm using here is the same as in https://github.com/oneapi-src/unified-runtime/pull/1127 (the hip implementation of this entrypoint is very similar).

Probably there should be only one macro. Any opinions about this @martygrant ?

Reply:
Same comment as @aarongreig.

Also I am not too sure if there is general style we follow (maybe threre should be) but aim to throw where the error is a legitimate failure, usually on most driver API calls unless it's just a hint where we ignore the error or treat it as a warning. I think in this specific case we really just want to return the error/result code if `cuda2urFormat` is not successful (invalid convertion) which is what you have done, but since we are already in a try-catch block that handles the thrown exception it is fine to throw via `UR_CHECK_ERROR`.

Reply:
Sorry @fabiomestre I didn't see this before. I do know that it's an objective for us to remove all abort/die calls. Unless I'm mistaken `UR_CHECK_ERROR` doesn't actually throw anything? I think we do probably need to have a discussion about correct error handling.

Reply:
Sorry for the delay in addressing these comments.

UR_CHECK_ERROR does throw. But I guess the pattern in the CUDA adapter is to do a try/catch in every entrypoint to catch those throws. I'm not a big fan of that pattern but that's not something that should be addressed in this PR.

I have updated the PR to use only UR_CHECK_ERROR. Would appreciate if you could review it again @aarongreig / @GeorgeWeb